### PR TITLE
chore: Unify api decorator

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

remove `model_api_key_required`, use `api_keys_required`

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by lightaime — COMMENTED** [review]
Thanks Wendong! This is the code cleaner. Only concern is what if the api key is sent to the client directly? We may need to change `api_keys_required` a bit.

We should also refactor the code with `dependencies_required`.

**Comment by zechengz** [comment]
I think we should make api key required and modules required separate

**Comment by Wendong-Fan** [comment]
> I think we should make api key required and modules required separate

Hey @lightaime , I though what you meant is to refactor `dependencies_required ` and `api_keys_required ` to unify these 2 decorators together, could you please help to clarify it? Thanks!

**Comment by lightaime** [comment]
> > I think we should make api key required and modules required separate
> 
> Hey @lightaime , I though what you meant is to refactor `dependencies_required ` and `api_keys_required ` to unify these 2 decorators together, could you please help to clarify it? Thanks!

Hi Wendong. I meant do it with a separate PR. Sorry for the confusion.

File: test/utils/test_commons.py [hunk 1]
```diff
@@ -70,19 +70,25 @@ def test_api_keys_required():
...
-        mock_api_keys_not_exist()
+    mock_instance = MockClass()
 
+    # Test case where all required API keys are present
+    assert mock_instance.mock_api_keys_exist() is True
+
+    # Test case where some required API keys are missing
+    with pytest.raises(ValueError) as exc:
+        mock_instance.mock_api_keys_not_exist()
     assert "Missing API keys: API_KEY_3" in str(exc.value)
```
Comment (Appointat) [inline test/utils/test_commons.py:93]:
```suggestion
@pytest.fixture
def setup_env_vars():
    original_env = os.environ.copy()
    os.environ['API_KEY_1'] = 'API_KEY_1_VALUE'
    os.environ['API_KEY_2'] = 'API_KEY_2_VALUE'
    yield
    os.environ.clear()
    os.environ.update(original_env)

def test_api_keys_required(setup_env_vars):
    class MockClass:
        @api_keys_required('API_KEY_1', 'API_KEY_2')
        def mock_api_keys_exist(self):
            return True

        @api_keys_required('API_KEY_1', 'API_KEY_2', 'API_KEY_3')
        def mock_api_keys_not_exist(self):
            return True

    mock_instance = MockClass()

    # Test case where all required API keys are present
    assert mock_instance.mock_api_keys_exist() is True

    # Test case where some required API keys are missing
    with pytest.raises(ValueError) as exc:
        mock_instance.mock_api_keys_not_exist()
    assert "Missing API keys: API_KEY_3" in str(exc.value)

    # Test case with no API keys set
    os.environ.clear()
    with pytest.raises(ValueError) as exc:
        mock_instance.mock_api_keys_exist()
    assert "Missing API keys: API_KEY_1, API_KEY_2" in str(exc.value)

def test_api_keys_required_empty(setup_env_vars):
    class MockClass:
        @api_keys_required()
        def mock_no_keys_required(self):
            return True

    mock_instance = MockClass()
    assert mock_instance.mock_no_keys_required() is True

def test_api_keys_required_non_existent(setup_env_vars):
    class MockClass:
        @api_keys_required('NON_EXISTENT_KEY')
        def mock_non_existent_key(self):
            return True

    mock_instance = MockClass()
    with pytest.raises(ValueError) as exc:
        mock_instance.mock_non_existent_key()
    assert "Missing API keys: NON_EXISTENT_KEY" in str(exc.value)
```

Comment (Appointat) [reply test/utils/test_commons.py:93]: It is just an optional suggestion.

Comment (Wendong-Fan) [reply test/utils/test_commons.py]: thanks Zikang!

**Pull request merged**